code/script.js

```diff
@@ -218,74 +257,196 @@ const generateBoard = () => {
+// Ylva - setting the currentQuestion object when you select something in the dropdown - "Does the person have..."
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  // The category variable is used to store what option group (category) the question belongs to.
+  const category = questions.options[questions.selectedIndex].parentNode.label;
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // Ylva - this is a variable that stores the actual value of the question we've selected.
+  const value = questions.value;
+  // this above could also be written as:
+  // const value = questions.options[questions.selectedIndex].value;
 
+  // Ylva - This below is variables that stores the actual value of the current questions we've selected.
   currentQuestion = {
-    category: category,
-    // value: value
-  }
-}
+  category: category,
+  value: value,
+  };
+
+ // Ylva - to check the values of the currentQuestion object
+console.log(currentQuestion);
+};
+
+const playerNameSave = () => {
+playerName = document.getElementById("nameInput").value;
+console.log(playerName);
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
+  const { category, value } = currentQuestion;
+//if-statement limits the number of questions allowed to 3 questions in total
+  if (numberOfQuestions > 0) {
+    console.log(numberOfQuestions);
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
+  // See if we should keep or remove people based on that (true or false)
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "Hair" || category === "Eyes") {
+   if (secret[category] === value) {
+    filterCharacters(true); //keep everyone with true according to hair/eye color
+   } else {
+   filterCharacters(); //remove everyone with false according to hair/eye color
+   }
+  } else if (category === "Accessories" || category === "Other") {
+if (secret[category].includes(value)) {
+filterCharacters(true); //keep everyone with true according to accessories/other
+} else {
+  filterCharacters(); //remove everyone with false according to accessories/other
   }
 }
+//don't know if there's something wrong here with the curly brackets above - still to check
+numberOfQuestions--;
+    console.log(numberOfQuestions);
+    numberOfQuestionsAllowed.innerText = `Number of questions left: ${numberOfQuestions}`;
+  } else if ((numberOfQuestions === 1)) {
```

Comment:
You don't need this. At line 333, you have a if numberOfGuesses is larger than zero then do the guessing. You could add an "else" after that whole code block and have it show the game over-screen then. (Unless you want something specific to change when you only have one guess left)

Reply:
@edv-rd Thanks - nice input!
So you mean I can add an "else" after line 333?
How would you write it...?